File: mobile/lib/supabase_config.dart
/// Адрес проекта и публикуемый ключ.
///
/// Эти два значения ПУБЛИЧНЫ по устройству Supabase: в вебе они уходят
/// в браузер любого посетителя, в приложении — лежат в бинаре, который
/// распаковывается за минуту. Защитой они не являются ни там, ни там.
/// Единственная защита данных — политики RLS (см. `supabase/migrations`).
/// Поэтому держать их в репозитории нормально, а прятать — самообман.
///
/// Ровно то же решение и та же причина, что в `lib/supabase/config.ts`
/// у веба: два клиента ходят в одну базу и живут по одним правилам.
///
/// Сервисный ключ (service_role) сюда не попадает НИКОГДА: он обходит RLS.
/// В приложении ему места нет в принципе — у него нет серверной части,
/// куда его можно было бы спрятать. См. CLAUDE.md, правило 3.
///
/// Значения перекрываются при сборке, если понадобится другой стенд:
///   flutter run --dart-define=SUPABASE_URL=... --dart-define=SUPABASE_KEY=...
class SupabaseConfig {
  const SupabaseConfig._();

  static const url = String.fromEnvironment(
    'SUPABASE_URL',
    defaultValue: 'https://jobvstdwoyifspaiwazn.supabase.co',
  );

  static const publishableKey = String.fromEnvironment(
    'SUPABASE_KEY',
    defaultValue: 'sb_publishable_9WvSKed5ELlCfrno_L3D-Q_L5rH4Wtk',
  );
}

File: shared/supabase-config.ts
// Адрес проекта и публикуемый ключ. ОДИН источник на оба приложения.
//
// Эти два значения ПУБЛИЧНЫ по устройству Supabase: они уходят в браузер
// любого посетителя и в бандл мобильного приложения, и защитой не являются.
// Единственная защита данных — политики RLS (см. supabase/migrations).
// Поэтому держать их в репозитории нормально, а прятать — самообман,
// создающий ложное чувство безопасности.
//
// Сервисный ключ (service_role) сюда не попадает НИКОГДА: он обходит RLS
// и живёт только в переменных окружения серверной части. См. CLAUDE.md, правило 3.
//
// ПОЧЕМУ ДВА ПРЕФИКСА. Сборщики подставляют значения переменных в код на
// этапе сборки, и каждый признаёт только свой префикс: Next — `NEXT_PUBLIC_`,
// Expo — `EXPO_PUBLIC_`. Тот, что не его, остаётся `undefined` и молча
// пропускается. Это не два источника правды: значение одно, читается оно
// в одном файле, а строк две ровно потому, что бандлера два.

export const SUPABASE_URL =
  process.env.NEXT_PUBLIC_SUPABASE_URL ??
  process.env.EXPO_PUBLIC_SUPABASE_URL ??
  'https://jobvstdwoyifspaiwazn.supabase.co'

export const SUPABASE_PUBLISHABLE_KEY =
  process.env.NEXT_PUBLIC_SUPABASE_PUBLISHABLE_KEY ??
  process.env.EXPO_PUBLIC_SUPABASE_PUBLISHABLE_KEY ??
  'sb_publishable_9WvSKed5ELlCfrno_L3D-Q_L5rH4Wtk'
